pyk/src/tests/integration/proof/test_custom_step.py

```diff
@@ -0,0 +1,258 @@
+            return Step(new_cterm, 1, (), ['CUSTOM-STEP.c.d'], cut=True)
+        return None
+
+
+CUSTOM_STEP_TEST_DATA_APPLY: Iterable[tuple[str, CTerm, KCFGExtendResult | None]] = (
+    (
+        'None',
+        CTerm(
+            config=KApply(
+                label=KLabel(name='<generatedTop>', params=()),
+                args=(
+                    KApply(
+                        label=KLabel(name='<k>', params=()),
+                        args=(
+                            KSequence(
+                                items=(KApply(label=KLabel(name='a_CUSTOM-STEP-SYNTAX_Step', params=()), args=()),)
+                            ),
+                        ),
+                    ),
+                    KApply(
+                        label=KLabel(name='<generatedCounter>', params=()),
+                        args=(KVariable(name='GENERATEDCOUNTER_CELL_c84b0b5f', sort=KSort(name='Int')),),
+                    ),
+                ),
+            ),
+            constraints=(),
+        ),
+        None,
+    ),
+    (
+        'Step',
+        CTerm(
+            config=KApply(
+                label=KLabel(name='<generatedTop>', params=()),
+                args=(
+                    KApply(
+                        label=KLabel(name='<k>', params=()),
+                        args=(
+                            KSequence(
+                                items=(KApply(label=KLabel(name='c_CUSTOM-STEP-SYNTAX_Step', params=()), args=()),)
+                            ),
+                        ),
+                    ),
+                    KApply(
+                        label=KLabel(name='<generatedCounter>', params=()),
+                        args=(KVariable(name='GENERATEDCOUNTER_CELL_c84b0b5f', sort=KSort(name='Int')),),
+                    ),
+                ),
+            ),
+            constraints=(),
+        ),
+        Step(
+            CTerm(
+                config=KApply(
+                    label=KLabel(name='<generatedTop>', params=()),
+                    args=(
+                        KApply(
+                            label=KLabel(name='<k>', params=()),
+                            args=(
+                                KSequence(
+                                    items=(KApply(label=KLabel(name='d_CUSTOM-STEP-SYNTAX_Step', params=()), args=()),)
+                                ),
+                            ),
+                        ),
+                        KApply(
+                            label=KLabel(name='<generatedCounter>', params=()),
+                            args=(KVariable(name='GENERATEDCOUNTER_CELL_c84b0b5f', sort=KSort(name='Int')),),
+                        ),
+                    ),
+                ),
+                constraints=(),
+            ),
+            1,
+            (),
+            ['CUSTOM-STEP.c.d'],
+            cut=True,
+        ),
+    ),
+)
+
+
+class TestCustomStep(KCFGExploreTest, KProveTest):
```

Comment:
Consider making this a `CTermSymbolicTest`, and moving the `create_kcfg_explore` fixture here. We can expose the fixture in the `pyk.testing` library once there are more tests requiring it.